content/technotes/2020-02-26-tic-ghactions.Rmd

```diff
@@ -0,0 +1,170 @@
+---
+slug: "tic-ghactions"
+title: Supercharge your GitHub Actions Experience with tic
+# Delete the package_version line below if your post is not about a package
+package_version: 0.5.0
+authors:
+  - Patrick Schratz
+# Set the date below to the publication date of your post
+date: 2020-02-26
+# Set categories to technotes if this is a tech note
+categories: technotes
+# Leave topicid blank below; will be set by editor
+topicid:
+# Minimal tags for a post about a community-contributed package 
+# that has passed software peer review are listed below
+# Consult the Technical Guidelines for information on choosing other tags
+tags:
+  - R
+  - GitHub
+  - GitHub Actions
+  - Continuous Integration
+# The summary below will be used by e.g. Twitter cards
+description: "tic aims to simplify CI related tasks for R projects (not just packages by the way) and now provides supercharged support for GitHub Actions. By enhancing the good work of [r-lib/actions](https://github.com/r-lib/actions) tic provides a DSL for R for CI tasks."
+---
+
+"Continuous Integration" (CI) has become a standard for proper software development.
+Checking the integrity of software after changes have been made is essential to ensure its proper functionality.
+Also, CI helps catching problems introduced by dependencies early when executed on a regular basis (usually done via scheduled [CRON](https://tecadmin.net/crontab-in-linux-with-20-examples-of-cron-schedule/) runs).
+
+Multiple professional providers exist ([Travis CI](https://travis-ci.com/), [Appveyor CI](https://www.appveyor.com/), [Circle CI](https://circleci.com/), etc.) which offer CI services to the public.
+Most services offer limited free build contingents for open-source projects.
+However, it has not been easy for the R community to jump on the train right away as most providers did not support the language natively on their side.
+Therefore, the R community started to create their own configurations to quickly automate the checking of R packages on [Travis CI](https://travis-ci.com/).
+In fact, the first commit of the official `r.rb` build script was made on [Feb 1st 2015 by Craig Citro](https://github.com/travis-ci/travis-build/commit/c697bb2240cfc1abb92a95f57d2e72c151104431).
+Since then 21 unique contributors maintained this configuration under the meta-lead of [Jim Hester](https://github.com/jimhester) and [Jeroen Ooms](https://github.com/jeroen) and many thousand R package maintainers jumped onto the train.
```

Comment:
```suggestion
Since then 21 unique contributors have maintained this configuration under the meta-lead of [Jim Hester](https://github.com/jimhester) and [Jeroen Ooms](https://github.com/jeroen) which allowed many thousands of R package maintainers to jump onto the train.
```